### ISSUE TITLE
Optimize UI for vertical space on desktops / landscape

### ISSUE THREAD

**Issue by blamarche** [jkk/shinkgs#122]
Currently almost 20% of vertical space is taken by navigation and decoration that would ideally, on a desktop / landscape layout, be instead filled by the actual board. Screenshots and mock at the end of the post.

Also, an option to hide the board coordinates would save some space as well and allow for larger boards. I don't mind pitching in to address these items, but would like your guidance on how you'd prefer it implemented and if these kinds of changes would be accepted in the first place.

Thanks!

![sshot](https://cloud.githubusercontent.com/assets/623830/25812266/458df8b2-3405-11e7-88a1-fecc6803f78d.jpg)

![sshot-mock](https://cloud.githubusercontent.com/assets/623830/25812268/499904ce-3405-11e7-85b0-7622bfe727fe.jpg)

**Comment by jkk** [comment]
I agree in principle about maximizing board size, but any implementation has to balance that with usability and aesthetics.

There's an issue for adding board view options: https://github.com/jkk/shinkgs/issues/95. Full screen mode and coordinate toggle are both definitely features I'd welcome.

For the default view, not crazy about moving the top-level nav. Moving the slider and prev/next nav to the sidebar may be possible. There may also be a way to present the back button and more menu more compactly. I tried some variations of that before but didn't find a great solution.

Open to considering layout revisions to the default but the devil is in the details and I can't promise an easy path to getting merged 🙂

**Comment by jkk** [comment]
One approach to changing the layout:

<img width="1043" alt="screen shot 2017-05-08 at 1 18 45 pm" src="https://cloud.githubusercontent.com/assets/5173/25816245/794a62c6-33f1-11e7-9cac-cb46dcb6d603.png">

Unfortunately this may require some non-trivial refactoring, since the nav is currently somewhat coupled to the board. This is motivation to fix that sooner rather than later.

**Comment by blamarche** [comment]
The other thing that I think would help is going with a vertical nav on the left side instead of a horizontal top bar (again better for desktop/landscape view only). 

Similar to how its done here, but it could still have labels under the icons:

![serveimage](https://cloud.githubusercontent.com/assets/623830/25817534/a82da928-33f5-11e7-9308-78ac1e30acae.jpg)

**Comment by jkk** [comment]
Maybe. Can't say I'm a fan of that style of navigation. It's fairly uncommon on the web, and implementing it in a way that works responsively is not a small amount of work. But still open to considering revisions if the details are fleshed out.

**Comment by blamarche** [comment]
Certainly its preference, though I would argue the vertical nav is fairly common in mobile apps and most 'admin' interfaces on the web. There's obviously a difference between document-based site navigation (ie. a blog, for example) and web app navigation. The navigation for this app is so simple I think it could pretty easily be moved to a sidebar or have a collapsible option.

For me, minimizing eye strain when playing go on a computer is a major priority. This is why I prefer to view as large of a board as possible. A real go board is roughly 18x18". On a 16:9 monitor it takes a pretty large screen to get even half of that in vertical space. The board is the most important element of a go playing interface, so finding ways to minimize the other on-screen elements during gameplay helps with that.

Regardless, really like work you've done so far.

**Comment by jkk** [comment]
Agreed. My preference is to make the UI intuitive and discoverable, favoring new and casual players over power users, with carefully-considered options as needed.

A full-screen mode is one relatively easy way to accomplish the "max board size" goal. I do want to make improvements to the default view - that just might take more time and thought. Leaning towards moving the slider nav to the side at the very least.

**Comment by blamarche** [comment]
Awesome, well let me know if you get to a point where you've decided how you want to move forward and  want help implementing some of it. Happy to pitch in when I can.

**Comment by jkk** [comment]
Playing with desktop layout ideas based on your suggestion...

<img width="1250" alt="screen shot 2017-05-08 at 4 46 09 pm" src="https://cloud.githubusercontent.com/assets/5173/25824653/287200a8-340e-11e7-9c8b-487a10cfa35b.png">

**Comment by blamarche** [comment]
Nice, I like where thats heading

**Comment by jkk** [comment]
Pushed some changes that are a first step in this direction, minus the side nav, which I'm still on the fence about.

<img width="1381" alt="screen shot 2017-05-22 at 6 46 37 pm" src="https://cloud.githubusercontent.com/assets/5173/26331514/48904a90-3f1f-11e7-9165-41063d685f90.png">

**Comment by blamarche** [comment]
Nice, thats a good improvement! That top nav still bugs me, personally. If the padding was smaller or there was a way to collapse/hide it, that'd keep the nav horizontal but still allow the screen space to be reclaimed.

**Comment by blamarche** [comment]
Nice progress on this project, its very nice! Would you like me to add a hide/expand nav button?

**Comment by blamarche** [comment]
Another option could be to add a fullscreen button, and use the :fullscreen css selector to remove nav elements in fullscreen only

**Comment by blamarche** [comment]
Quick example:

Unfortunately currently the API is prefixed, but pretty easy to work around. Use mozRequestFullScreen() on firefox, webkitRequestFullscreen() for chrome, msRequestFullscreen() for IE/edge. Whats very nice is that almost no other changes are needed for this to work. After entering fullscreen, clicking 'back' or any other navigation will take the site out of fullscreen and resume normal functionality.

JS:

```
document.querySelector('.GameScreen').webkitRequestFullscreen();
document.addEventListener("fullscreenerror", function( event ) {
  alert('You need to grant fullscreen permission to enable this button.');
}
```

CSS:

```
.GameScreen:fullscreen {
  top: 0px;
}
```
